first-party/jib-extension-3147/src/main/java/com/google/cloud/tools/jib/extension/e3147/ClasspathMainClassExtension.java

```diff
@@ -0,0 +1,71 @@
+package com.google.cloud.tools.jib.extension.e3147;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.cloud.tools.jib.api.buildplan.ContainerBuildPlan;
+import com.google.cloud.tools.jib.gradle.extension.GradleData;
+import com.google.cloud.tools.jib.gradle.extension.JibGradlePluginExtension;
+import com.google.cloud.tools.jib.maven.extension.JibMavenPluginExtension;
+import com.google.cloud.tools.jib.maven.extension.MavenData;
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger;
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger.LogLevel;
+import com.google.cloud.tools.jib.plugins.extension.JibPluginExtensionException;
+
+public class ClasspathMainClassExtension implements JibGradlePluginExtension<Void>, JibMavenPluginExtension<Void> {
+
+	@Override
+	public Optional<Class<Void>> getExtraConfigType() {
+		return Optional.empty();
+	}
+
+	@Override
+	public ContainerBuildPlan extendContainerBuildPlan(ContainerBuildPlan buildPlan, Map<String, String> properties,
+			Optional<Void> extraConfig, GradleData gradleData, ExtensionLogger logger)
+			throws JibPluginExtensionException {
+		setEnvsFromEntrypoint(buildPlan, logger);
+		return buildPlan;
+	}
+
+	@Override
+	public ContainerBuildPlan extendContainerBuildPlan(ContainerBuildPlan buildPlan, Map<String, String> properties,
+			Optional<Void> extraConfig, MavenData mavenData, ExtensionLogger logger)
+			throws JibPluginExtensionException {
+		setEnvsFromEntrypoint(buildPlan, logger);
+		return buildPlan;
+	}
+
+	/*
+	 * see com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor.computeEntrypoint(RawConfiguration, ProjectProperties)
+	 */
+	private void setEnvsFromEntrypoint(ContainerBuildPlan buildPlan, ExtensionLogger logger) {
```

Comment:
Could you add some tests for it? Probably you'll want
```suggestion
      @VisibleForTesting
	void setEnvsFromEntrypoint(ContainerBuildPlan buildPlan, ExtensionLogger logger) {
```
so that you can call this method from tests. For the general idea about setting up tests, see other first-party extensions.